app/models/session.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# Contains a token to identify users
+class Session
+  include ActiveModel::Validations
+  include ActiveModel::Serialization
+
+  def initialize(session_params)
+    @email = session_params[:email]
+    @password = session_params[:password]
+  end
+
+  validates :email, presence: true
+  validates :password, presence: true
+  validate :authenticated
+
+  def token
+    JwtCodec.new.encode({ id: user.id })
+  end
+
+  private
+
+  attr_reader :email, :password
+
+  def user
+    @user ||= User.find_by(email: @email)&.authenticate(@password)
+  end
```

Comment:
Don't you want to create separate PORO class to make this method look like
```suggestion
  def user
    @user ||= Users::Authenticate(email: @email, password: @password)
  end
```

Reply:
Thanks, I like it. Will apply your suggestion in a future PR.